core/modules.rs

```diff
@@ -292,6 +292,46 @@ impl ModuleLoader for NoopModuleLoader {
   }
 }
 
+pub struct InternalModuleLoader;
+
+impl ModuleLoader for InternalModuleLoader {
+  fn resolve(
+    &self,
+    specifier: &str,
+    referrer: &str,
+    kind: ResolutionKind,
+  ) -> Result<ModuleSpecifier, Error> {
+    dbg!(specifier, referrer, &kind);
+    let specifier = ModuleSpecifier::parse(specifier).unwrap();
+    if kind == ResolutionKind::Import {
+      let referrer_specifier = ModuleSpecifier::parse(referrer).ok();
+      if specifier.scheme() == "deno" {
+        return if referrer == "." || referrer_specifier.unwrap().scheme() == "deno" {
```

Comment:
<3

Reply:
fyi the deno specifier being used here is just temporary, and will be replaced with `internal`

Reply:
Oh yeah, I wasn't even thinking of the proposed `deno:` imports but was just happy to see that loading the internals was indeed blocked nicely.

`internal` does make more sense. I guess the only downside is that users are probably somewhat used to seeing `deno:/...` paths in error messages.